framework/include/math_utils.hpp

```diff
@@ -12,6 +12,10 @@ namespace gvk
 	/// extract the translation part out of a matrix
 	extern glm::vec3 get_translation_from_matrix(const glm::mat4& m);
 
+	glm::mat4 cancel_translation_from_matrix(const glm::mat4& m);
+
+	glm::mat4 mirror_matrix(const glm::mat4& m, const glm::length_t axis = 0);
```

Comment:
Please add comments in the following style:
 ```
/** Function description
 *   @param  aMatrix   Parameter description
 *   @param  aAxis     Parameter description
 *   @return Return value description
 */
glm::mat4 mirror_matrix(const glm::mat4& aMatrix, const glm::length_t aAxis = 0);
```
I know that most of the functions in `math_utils.hpp` are not well-behaved in that regard, so please don't let yourself be guided by them but follow the coding guidelines!

Reply:
Hmm, a small `enum struct principal_axis : uint32_t { x = 0u, y, z };`  would be better for the "axis" parameter since then only the well defined values can be passed to the function, it becomes more clear to the user, and the code becomes more readable..